hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CustomFeesValidator.java

```diff
@@ -228,20 +237,48 @@ private void validateFixedFeeForCreation(
         }
     }
 
+    private void validateFractionalFeeForCreation(TokenType tokenType, CustomFee fee, Set<CustomFee> fees) {
+        validateTrue(isFungibleCommon(tokenType), CUSTOM_FRACTIONAL_FEE_ONLY_ALLOWED_FOR_FUNGIBLE_COMMON);
+
+        final var fractionalFee = fee.fractionalFee();
+        final var numerator = fractionalFee.fractionalAmount().numerator();
+        final var denominator = fractionalFee.fractionalAmount().denominator();
+        final var minimumUnitsToCollect = fractionalFee.minimumAmount();
+        final var nominalMax = fractionalFee.maximumAmount();
+        final var maximumUnitsToCollect = nominalMax == 0 ? Long.MAX_VALUE : nominalMax;
+
+        validateTrue(denominator != 0, FRACTION_DIVIDES_BY_ZERO);
+        validateTrue(numerator > 0 && denominator > 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        validateTrue(minimumUnitsToCollect >= 0 && maximumUnitsToCollect >= 0, CUSTOM_FEE_MUST_BE_POSITIVE);
+        validateTrue(maximumUnitsToCollect >= minimumUnitsToCollect, FRACTIONAL_FEE_MAX_AMOUNT_LESS_THAN_MIN_AMOUNT);
+
+        fees.add(fee);
```

Comment:
same here, we should not be adding this to fees, since that list is only needed for collector auto association when token mentioned is a sentinel value.